internal/services/loganalytics/log_analytics_linked_storage_account_resource.go

```diff
@@ -142,7 +142,7 @@ func resourceLogAnalyticsLinkedStorageAccountRead(d *pluginsdk.ResourceData, met
 
 		dataSourceType := ""
 		if props.DataSourceType != nil {
-			dataSourceType = string(*props.DataSourceType)
+			dataSourceType = strings.ToLower(string(*props.DataSourceType))
```

Comment:
and here

Reply:
The symptom is that there is tf diff.
After investigated, I found tf only allows lower case but API always returns the value with the capital initial. So it causes tf diff. I've updated PR to make the value returned by API to be lower case. Please take another look. Thanks.

Reply:
No we can't, but we should support the correct casing. What we should do change the validation to be the correct case, save the correct casing in state, and flag case insensitivity validatiuon and diff ignore with the 4.0 flag so for 4.0 we go from ignoring case to respecting it? and solve the actually issue here

Reply:
@katbyte , thanks for the suggestion. Updated PR. It includes below two main changes.

One change is setting ignoreCase of ValidateFunc to true to allow lower case in tf config to avoid breaking change.
Another change is adding DiffSuppress to ignore the difference between the value in state file (ex: "CustomLogs") and the value in tf config (ex: "customlogs").